crates/nargo_cli/benches/iai.rs

```diff
@@ -0,0 +1,27 @@
+//! Select representative tests to bench with iai
+use assert_cmd::prelude::{CommandCargoExt, OutputAssertExt};
+use iai::black_box;
+use paste::paste;
+use std::process::Command;
+include!("./utils.rs");
+
+macro_rules! iai_command {
+    ($command_name:tt, $command_string:expr) => {
+        paste! {
+            fn [<iai_selected_tests_ $command_name>]() {
+                let test_program_dirs = get_selected_tests();
+                for test_program_dir in test_program_dirs {
+                    let mut cmd = Command::cargo_bin("nargo").unwrap();
+                    cmd.arg("--program-dir").arg(&test_program_dir);
+                    cmd.arg($command_string);
+
+                    black_box(cmd.assert());
+                }
+            }
+        }
+    };
+}
+iai_command!(execution, "execute");
+iai_command!(prove, "prove");
```

Comment:
I think we don't want to benchmark the proving time since that should be constant with respect to Noir -- Its mainly the responsibility of the backend, though if this does not make the benchmarks single threaded and does not add a lot code, perhaps its fine to leave for now